### ISSUE TITLE
az ssh vm: Using additional ssh arguments broken in 2.40.0

### ISSUE THREAD

**Issue by JWohlschlag-Euclid** [Azure/azure-cli#24058]
## Describe the bug

**Command Name**
`az ssh vm`

**Errors:**
```
az ssh vm: 'exitonforwardfailure=yes' is not a valid value for '--output'. Allowed values: json, jsonc, yaml, yamlc, table, tsv, none.

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az ssh vm --resource-group {} --name {} -- -f -N -o {} -L {} -L {}`

## Expected Behavior
The command reference guide shows adding extra SSH arguments as the following:

Using additional ssh arguments
az ssh vm --ip 1.2.3.4 -- -A -o ForwardX11=yes

As of azure-cli 2.40.0, this command reference guide example produces a similar error as what is shown above. In azure-cli 2.39.0, adding extra SSH arguments via -- -o worked properly. As of azure-cli 2.40.0, using -- -o produces an error that the output is of an unexpected format.

## Environment Summary
```
Windows-10-10.0.22621-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.40.0

Extensions:
ssh 1.1.1

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```
## Additional Context
Adding extra SSH arguments via az ssh vm -- -o worked in 2.39.0, but is broken in 2.40.0.

**Comment by yonzhan** [comment]
route to CXP team

**Comment by JWohlschlag-Euclid** [comment]
This issue has been resolved in Azure CLI version 2.41.0.
